### ISSUE TITLE
React web app is not working on IE 10

### ISSUE THREAD

**Issue by Plash-jindal** [facebook/react#11429]
**Do you want to request a *feature* or report a *bug*?**
Report a bug

**What is the current behavior?**
Web app is displaying black screen on load.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar (template for React 16: https://jsfiddle.net/Luktwrdm/, template for React 15: https://jsfiddle.net/hmbg7e9w/).**
I just created a demo app using create-react-app, and then I npm start that application. After that I tried to run it on IE 10, but it displays black screen error saying Expected identifier in console. 

**What is the expected behavior?**
The application should be displayed on the IE 10 browser but this is not happening. 

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**
"react": "^16.0.0",
    "react-dom": "^16.0.0",
    "react-scripts": "1.0.15"

Browser: IE 10

**Comment by mAiNiNfEcTiOn** [comment]
@Plash-jindal this was already fixed on this PR: https://github.com/facebook/react/pull/11421

But it is waiting release.

It only affects react.dom.development ... if you use react.dom.production should work :)

**Comment by Plash-jindal** [comment]
But it is working in IE 11. 
It’s just the older versions where it’s not working.

**Comment by mAiNiNfEcTiOn** [comment]
@Plash-jindal must be on your IE 11, on mine your JS Fiddle doesn't work: https://ibb.co/ioMYfw

**Comment by mAiNiNfEcTiOn** [comment]
Maybe a better link would be: https://codesandbox.io/s/r5zo33o9m

**Comment by Plash-jindal** [comment]
Well, after running the demo app in production mode, it ran on the IE 10. 
Now I have to just check it with my project as I am using Java as the backend and react as frontend. Moreover, webpack is used for the deployment, so I will test accordingly in production mode for the IE 10 browser.

Thank you.

**Comment by mAiNiNfEcTiOn** [comment]
👍

**Comment by gaearon** [comment]
React `16.1.0-beta` has been released. Please update `react`, `react-dom`, and `react-test-renderer` (if you use it) to this version and let us know if it solved the issue! We’d appreciate if you could test before Monday when we plan to get `16.1.0` out.

**Comment by Plash-jindal** [comment]
@gaearon Hi, I have tested it using this versions but the application still not running on IE 10. 
It is giving me the same error.

**Comment by gaearon** [comment]
Can you please give more details? What line is it referring to?

**Comment by Plash-jindal** [comment]
Its in the file bundle.js, line 17271, character 2.

**Comment by gaearon** [comment]
That doesn't tell us much given that we don't have your bundle :-)

Please prepare an isolated project that only contains React and reproduces the issue.

**Comment by Plash-jindal** [comment]
Well, I just created a basic application using create-react-app, and then I tried to run it after updating the react and react-dom modules on IE 10. 
It is giving the error while following these steps.

**Note:** It is the basic startup application which is generated by using create-react-app.

**Comment by Plash-jindal** [comment]
Yes, I did that. 
But one thing, when I update react-dom, it tell me the warning saying:

**UNMET PEER DEPENDENCY react@16.1.0-beta
└── react-dom@16.1.0-beta 

npm WARN react-dom@16.1.0-beta requires a peer of react@^16.0.0 but none was installed.**

**Comment by gaearon** [comment]
I just created a new app, installed `react` and `react-dom` with `16.1.0-beta` version, and opened IE in IE10 mode. I see these errors:

```
'Map' is undefined
'Set' is undefined
```

I followed instructions in https://reactjs.org/docs/javascript-environment-requirements.html and that gave me a working app.

So we'll need more information about how you reproduced this. Maybe you can publish the project you created?

**Comment by Plash-jindal** [comment]
Sure, I will publish the project and you can check that.

**Comment by Plash-jindal** [comment]
https://jsfiddle.net/plash2908/sv6L2u3r/#&togetherjs=zE28dwAaU9

**Comment by Plash-jindal** [comment]
Please collaborate.

**Comment by gaearon** [comment]
I updated your fiddle. Does this work in IE10? https://jsfiddle.net/3scsxLtz/1/

**Comment by Plash-jindal** [comment]
[my-app 2.zip](https://github.com/facebook/react/files/1446713/my-app.2.zip)

**Comment by mAiNiNfEcTiOn** [comment]
I've updated my codesandbox: https://codesandbox.io/s/p2rkzl6xwq

Uses the _beta_ version and works fine on IE 11 (proof below)... Testing now with IE 10...

<img width="961" alt="screen shot 2017-11-06 at 16 18 50" src="https://user-images.githubusercontent.com/1002056/32448070-396d25d8-c30e-11e7-99a7-c4461f917f41.png">

**Comment by Plash-jindal** [comment]
@gaearon No, it doesn't work in IE 10.

**Comment by mAiNiNfEcTiOn** [comment]
Well... lol for codesandbox which doesn't run on IE 10 😄 

@gaearon I ran your jsfiddle and it has this problem:

<img width="961" alt="screen shot 2017-11-06 at 16 23 12" src="https://user-images.githubusercontent.com/1002056/32448319-e664b54e-c30e-11e7-87d2-1d392ae82caa.png">

**Comment by Plash-jindal** [comment]
@gaearon  @mAiNiNfEcTiOn  please check this, https://codesandbox.io/embed/811v7yxyjj

It is not working on IE 10.

**Comment by mAiNiNfEcTiOn** [comment]
@Plash-jindal for me the whole codesandbox.io doesn't work on IE 10 😉

**Comment by mAiNiNfEcTiOn** [comment]
@gaearon it seems that this line https://github.com/facebook/react/blob/master/packages/react/src/ReactElementValidator.js#L63 Is the problem.

**Comment by Plash-jindal** [comment]
@mAiNiNfEcTiOn  Please try to install my project which I uploaded it here as a zip file and do the npm install and try to run it on IE 10.

**Comment by gaearon** [comment]
As I wrote [above](https://github.com/facebook/react/issues/11429#issuecomment-342148046), **React requires Map and Set polyfills.** 😉 

Please follow these instructions to include them:
https://reactjs.org/docs/javascript-environment-requirements.html

**Comment by mAiNiNfEcTiOn** [comment]
@gaearon then, will do it on your fiddle too to check and confirm. 👍

**Comment by Plash-jindal** [comment]
@gaearon I tried that, it did not work either.

**Comment by gaearon** [comment]
I'll be looking at your zip in a sec.

**Comment by Plash-jindal** [comment]
Sure, thanks. @gaearon

**Comment by mAiNiNfEcTiOn** [comment]
I used `polyfill.io` with its `default-3.6` configuration which polyfills Map and Set, doesn't seem to work. Maybe _jsfiddle_ doesn't support IE10 too? (GIving an error in the `_display` frame 🙄 )

<img width="961" alt="screen shot 2017-11-06 at 16 36 28" src="https://user-images.githubusercontent.com/1002056/32449005-bcc6883c-c310-11e7-913c-333ddbc9da09.png">

**Comment by gaearon** [comment]
Yes, I've verified JSFiddle itself is broken so we shouldn't test in it.

**Comment by mAiNiNfEcTiOn** [comment]
@Plash-jindal I tested your stuff on IE 10... works:

<img width="961" alt="screen shot 2017-11-06 at 16 42 50" src="https://user-images.githubusercontent.com/1002056/32449347-98fbd5a0-c311-11e7-9330-d9fd9ca57c64.png">

**Comment by gaearon** [comment]
I have verified the zipped project works on a real IE10 instance (via BrowserStack).

![img_0224](https://user-images.githubusercontent.com/810438/32449528-b2fbaa0a-c309-11e7-9908-1dcee8b3a6e2.JPG)

**Comment by Plash-jindal** [comment]
Are you sure that you are not running this on production environment?

**Comment by mAiNiNfEcTiOn** [comment]
@Plash-jindal i unzipped, did `npm i` and did `npm start`. As you can see I accessed http://localhost:3000 ... worked just fine.

**Comment by Plash-jindal** [comment]
![screen shot 2017-11-06 at 16 29 24](https://user-images.githubusercontent.com/16471729/32449729-a9e8b648-c312-11e7-83b4-2bb996f3eca0.png)


I get this error on IE 10 when I try to run it.

**Comment by mAiNiNfEcTiOn** [comment]
@Plash-jindal Look at the _Modo de explorador_ ... is set to IE8.

**Comment by Plash-jindal** [comment]
@mAiNiNfEcTiOn Yes, its working fine now. My mistake, I was running it with Explorer mode as IE 8. 👍

**Comment by luc-tuyishime** [comment]
Create React App is not displaying in internet explorer 11. How can i fix that?

**Comment by emreavsar** [comment]
Interested too!

**Comment by dmsps** [comment]
If it doesnt work in IE, try to use **babel-polyfill**

**Comment by mAiNiNfEcTiOn** [comment]
Or polyfill.io

**Comment by zxhhh11** [comment]
when I ran my react project ,my index.js file is like this
```
import 'core-js/es/map';
import 'core-js/es/set';
import 'raf/polyfill';
 import "core-js";
 import "regenerator-runtime/runtime";
import React from 'react';
import ReactDOM from 'react-dom';
```

but it doesn't work  in IE 10 , Who can help me tell me what to do ?
